src/citrine/builders/design_spaces.py

```diff
@@ -36,6 +41,14 @@ def enumerate_cartesian_product(
         description for the EnumeratedDesignSpace
 
     """
+    # Check that the grid size is small enough to not cause memory issues
+    grid_size = np.prod(
+        [len(grid_points) for grid_points in design_grid.values()]
```

Comment:
Here, the grid_size is the number of rows in `data`

Reply:
Changed to calculate total grid size